pkg/cmd/backuprestore/backupnoconfig.go

```diff
@@ -0,0 +1,177 @@
+		Run: func(cmd *cobra.Command, args []string) {
+			must := func(fn func() error) {
+				if err := fn(); err != nil {
+					if cmd.HasParent() {
+						klog.Fatal(err)
+					}
+					fmt.Fprint(backupNoConf.errOut, err.Error())
+				}
+			}
+
+			must(backupNoConf.Validate)
+			must(backupNoConf.Run)
+		},
+	}
+	backupNoConf.AddFlags(cmd.Flags())
+	return cmd
+}
+
+func (b *backupNoConfig) AddFlags(fs *pflag.FlagSet) {
+	b.backupOptions.AddFlags(fs)
+}
+
+func (b *backupNoConfig) Validate() error {
+	return b.Validate()
+}
+
+func (b *backupNoConfig) Run() error {
+	backupsClient, err := b.getBackupClient()
+	if err != nil {
+		return err
+	}
+
+	b.scheduler, _ = gcron.NewScheduler(
+		gcron.WithLimitConcurrentJobs(1, gcron.LimitModeWait),
+		gcron.WithGlobalJobOptions(
+			gcron.WithLimitedRuns(1),
+			gcron.WithSingletonMode(gcron.LimitModeWait)))
+	defer func() { _ = b.scheduler.Shutdown() }()
+
+	if err = b.extractBackupSpecs(backupsClient); err != nil {
+		return err
+	}
+
+	err = b.scheduleBackup()
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (b *backupNoConfig) getBackupClient() (backupv1client.BackupsGetter, error) {
+	kubeConfig, err := clientcmd.BuildConfigFromFlags("", b.kubeConfig)
+	if err != nil {
+		bErr := fmt.Errorf("error loading kubeconfig: %v", err)
+		klog.Error(bErr)
+		return nil, bErr
+	}
+
+	backupsClient, err := backupv1client.NewForConfig(kubeConfig)
+	if err != nil {
+		bErr := fmt.Errorf("error creating etcd backups client: %v", err)
+		klog.Error(bErr)
+		return nil, bErr
+	}
+
+	return backupsClient, nil
+}
+
+func (b *backupNoConfig) extractBackupSpecs(backupsClient backupv1client.BackupsGetter) error {
+	backups, err := backupsClient.Backups().List(context.Background(), v1.ListOptions{})
+	if err != nil {
+		lErr := fmt.Errorf("could not list backup CRDs, error was: [%v]", err)
+		klog.Error(lErr)
+		return lErr
+	}
+
+	if len(backups.Items) == 0 {
+		lErr := fmt.Errorf("no backup CRDs exist, found [%v]", backups)
+		klog.Error(lErr)
+		return lErr
+	}
+
+	idx := slices.IndexFunc(backups.Items, func(backup backupv1alpha1.Backup) bool {
+		return backup.Name == "default"
+	})
+	if idx == -1 {
+		sErr := fmt.Errorf("could not find default backup CR, found [%v]", backups.Items)
+		klog.Error(sErr)
+		return sErr
+	}
+
+	defaultBackupCR := backups.Items[idx]
+	b.schedule = defaultBackupCR.Spec.EtcdBackupSpec.Schedule
+	b.retention = defaultBackupCR.Spec.EtcdBackupSpec.RetentionPolicy
+
+	return nil
+}
+
+func (b *backupNoConfig) backup() error {
+	return backup(&b.backupOptions)
+}
+
+func (b *backupNoConfig) scheduleBackup() error {
+	if _, err := b.scheduler.NewJob(
+		gcron.CronJob(
+			b.schedule,
+			false,
+		),
+		gcron.NewTask(b.backup()),
+		gcron.WithEventListeners(
+			gcron.AfterJobRuns(
+				func(jobID uuid.UUID, jobName string) {
+					b.pruneBackups()
+				},
+			),
+		),
+	); err != nil {
+		return err
+	}
+
+	b.scheduler.Start()
```

Comment:
From the documentation I can see:
```
	// Start begins scheduling jobs for execution based
	// on each job's definition. Job's added to an already
	// running scheduler will be scheduled immediately based
	// on definition. Start is non-blocking.
	Start()
```

https://pkg.go.dev/github.com/go-co-op/gocron/v2#Scheduler

so this sidecar will continue to restart every time it runs, not sure how it is able to schedule a cronjob accordingly

Reply:
>  the job will be added to the scheduler **immediately**, and will be scheduled upon **definition**
iiuc, this means the job is being added, but the execution is based on the `crontab` 

why this will restart the whole sidecar ?

Reply:
there is no crontab here, the library will just sleep on a go channel :rofl:

Reply:
achsoooo .. but why the container will restart every time ?

Reply:
what's stopping it from exiting the command?

Reply:
achsooo .. ok  .. so either we run it a daemon or have a controller that invoke the cmd upon reconciliation, or ?

Reply:
just trying whether I'm stupid here, but this example immediately exits without running any task:

```
import (
	"fmt"
	gcron "github.com/go-co-op/gocron/v2"
)

func main() {
	scheduler, err := gcron.NewScheduler(
		gcron.WithLimitConcurrentJobs(1, gcron.LimitModeWait),
		gcron.WithGlobalJobOptions(
			gcron.WithLimitedRuns(1),
			gcron.WithSingletonMode(gcron.LimitModeWait)))
	if err != nil {
		panic(err)
	}

	_, err = scheduler.NewJob(gcron.CronJob("* * * * *", false), gcron.NewTask(func() {
		fmt.Printf("hello world\n")
	}))
	if err != nil {
		panic(err)
	}
	scheduler.Start()

}
```

Reply:
> achsooo .. ok .. so either we run it a daemon or have a controller that invoke the cmd upon reconciliation, or ?

well, hence my question on the other thread about the life cycle you're thinking about, because this command is not going to work :'D

Reply:
no u r right, its my bad :D .. 

but i followed the same approach as in the `requestBackup` and `prune` CMDs 

probably we need a controller that reacts only to the `default` CR and invoke the cmd 

wdyt